src/sentry/web/frontend/oauth_token.py

```diff
@@ -46,81 +49,95 @@ def post(self, request: HttpRequest) -> HttpResponse:
         client_id = request.POST.get("client_id")
         if not client_id:
             return self.error(request, "invalid_client", "missing client_id")
-        if grant_type == GrantTypes.AUTHORIZATION:
-            return self._get_access_tokens(request, client_id)
-        elif grant_type == GrantTypes.REFRESH:
-            return self._get_refresh_token(request, client_id)
-        else:
+        if grant_type not in [GrantTypes.AUTHORIZATION, GrantTypes.REFRESH]:
             return self.error(request, "unsupported_grant_type")
-
-    def _get_access_tokens(self, request, client_id):
-        redirect_uri = request.POST.get("redirect_uri")
-        code = request.POST.get("code")
-
         try:
             application = ApiApplication.objects.get(
                 client_id=client_id, status=ApiApplicationStatus.active
             )
         except ApiApplication.DoesNotExist:
             return self.error(request, "invalid_client", "invalid client_id")
 
+        # date that we started requiring client secrets
+        cutoff_date = datetime.datetime(2023, 7, 16).time()
+
+        if application.date_added.time() > cutoff_date:
```

Comment:
Let's remove the cutoff date check here. We should not maintain backwards compatibility here. The security concern takes precedence in this case.